renku_notebooks/api/notebooks.py

```diff
@@ -146,7 +147,8 @@ def launch_notebook(user):
 
     # only create a pull secret if the project has limited visibility and a token is available
     if config.GITLAB_AUTH and gl_project.visibility in {"private", "internal"}:
-        secret_name = f"{user.get('name')}-registry"
+        safe_username = escapism.escape(user.get("name"), escape_char="-").lower()
+        secret_name = f"{safe_username}-registry"
```

Comment:
Should this be handled inside `create_or_replace_registry_secret`? Or in a separate function? We'll need it again when removing the secret...

Reply:
It agree that once we tackle the removal, this logic will very likely end up in a separate helper function. But since it's not yet entirely clear to me how we will implement the removal (I think it would be nice to couple the lifetime of the secret as directly to the pod lifetime) I'd leave this for later.